cs/src/core/RuntimeDeserializer.cs

```diff
@@ -0,0 +1,671 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Bond
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using Bond.IO.Safe;
+    using Bond.Protocols;
+
+    /// <summary>
+    /// Deserializes <see cref="RuntimeObject"/>s from a protocol reader using only a runtime schema.
+    /// </summary>
+    public static class RuntimeDeserialize
+    {
+        /// <summary>
+        /// Deserializes a <see cref="RuntimeObject"/> from a tagged protocol reader using only a runtime schema.
+        /// </summary>
+        /// <param name="reader">The tagged protocol reader to read the object data from.</param>
+        /// <param name="schema">The schema to use when reading the object data.</param>
+        /// <returns>A <see cref="RuntimeObject"/> created from <paramref name="reader"/> and <paramref name="schema"/>.</returns>
+        public static RuntimeObject From(IClonableTaggedProtocolReader reader, RuntimeSchema schema)
+        {
+            return new RuntimeDeserializer(schema).Deserialize(reader);
+        }
+
+        /// <summary>
+        /// Deserializes a <see cref="RuntimeObject"/> from an untagged protocol reader using only a runtime schema.
+        /// </summary>
+        /// <param name="reader">The untagged protocol reader to read the object data from.</param>
+        /// <param name="schema">The schema to use when reading the object data.</param>
+        /// <returns>A <see cref="RuntimeObject"/> created from <paramref name="reader"/> and <paramref name="schema"/>.</returns>
+        public static RuntimeObject From(IClonableUntaggedProtocolReader reader, RuntimeSchema schema)
+        {
+            return new RuntimeDeserializer(schema).Deserialize(reader);
+        }
+    }
+
+    /// <summary>
+    /// Deserializes <see cref="RuntimeObject"/>s from a protocol reader using only a <see cref="RuntimeSchema"/>.
+    /// </summary>
+    public class RuntimeDeserializer
```

Comment:
I'd like to clarify my comment about an implementation based on `IProtocolWriter`. I don't think that would be a good general purpose implementation to contribute to Bond. But it could be a relatively simple solution for people who need similar functionality until we have a full, general purpose API in Bond.

As for implementation based on Bond parser/transform architecture, I concede that there is some learning curve and we actually need to provide more documentation/examples on how to do that.